apps/researcher/src/lib/route-segment-transformer.test.ts

```diff
@@ -0,0 +1,21 @@
+import {describe, expect, it} from '@jest/globals';
+import {
+  encodeRouteSegment,
+  decodeRouteSegment,
+} from './route-segment-transformer';
+
+describe('encodeRouteSegment', () => {
+  it('encodes a URI', () => {
```

Comment:
Minor: `encodes a route segment`? The segment could be a URI, but doesn't have to be?

Reply:
I renamed it to 'encodes a string with dots'